types/relay-runtime/relay-runtime-tests.tsx

```diff
@@ -94,7 +94,7 @@ const environment = new Environment({
             handle(field, record, argValues) {
                 if (
                     record != null &&
-                    record.__typename === ROOT_TYPE &&
+                    record.getType() === ROOT_TYPE &&
```

Comment:
think in practice this doesn't need to change, might need to let `ReadOnlyRecordProxy` extend an object?

Reply:
Yeah, definitely seems odd. The Flow types don't appear to include `__typename` or extend anything that would make this work differently. 🤔

Reply:
This doesn't work for me. I'm using :

```
"@types/relay-runtime": "14.1.6",
"@types/react-relay": "14.1.3",
"react-relay": "14.1.0",
"relay-runtime": "14.1.0",
```

I have a code that follows [the Relay documentation](https://relay.dev/docs/guided-tour/reusing-cached-data/filling-in-missing-data/):

```
const missingFieldHandlers = [
  {
    handle(field, record, argValues): ?string {
      if (
        record != null &&
        record.__typename === ROOT_TYPE &&
        field.name === 'user' &&
        argValues.hasOwnProperty('id')
      ) {
        // If field is user(id: $id), look up the record by the value of $id
        return argValues.id;
      }
      if (
        record != null &&
        record.__typename === ROOT_TYPE &&
        field.name === 'story' &&
        argValues.hasOwnProperty('story_id')
      ) {
        // If field is story(story_id: $story_id), look up the record by the
        // value of $story_id.
        return argValues.story_id;
      }
      return undefined;
    },
    kind: 'linked',
  },
];
```

The code fails in runtime because `record` does not have `getType` method. Is it possible the types are wrong or is it my code?